src/files.rs

```diff
@@ -0,0 +1,1044 @@
+    })?;
+    debug!("Waiting for threads to finish...");
+    // Wait the termination of all consumers
+    for handle in handlers {
+        handle.join()??;
+    }
+    if let Err(_e) = sender_composer.send(None) {
+        return Err(Error::SenderError());
+    }
+    let mut files_ignored = cfg.files_ignored.lock()?;
+    let mut res = cfg.res.lock()?;
+    let project_coverage = covs
+        .get(&("PROJECT_ROOT".to_string()))
+        .ok_or(Error::HashMapError())?
+        .coverage;
+    // Get final  metrics for all the project
+    let composer_output = composer.join()??;
+    let project_metric = FileMetrics::new(
+        get_project_metrics(composer_output, Some(project_coverage))?,
+        "PROJECT".to_string(),
+        "-".to_string(),
+    );
+    files_ignored.sort();
+    res.sort_by(|a, b| a.file.cmp(&b.file));
+    // Get AVG MIN MAX and complex files
+    let complex_files = res
+        .iter()
+        .filter(|m| m.metrics.is_complex)
+        .cloned()
+        .collect::<Vec<FileMetrics>>();
+    let m = res
+        .iter()
+        .map(|metric| metric.metrics)
+        .collect::<Vec<Metrics>>();
+    let (avg, max, min) = get_cumulative_values(&m);
+    res.push(project_metric);
+    res.push(FileMetrics::avg(avg));
+    res.push(FileMetrics::max(max));
+    res.push(FileMetrics::min(min));
+    Ok((
+        (*res).clone(),
+        (*files_ignored).clone(),
+        complex_files,
+        project_coverage,
+    ))
+}
+
+/// Prints the the given  metrics ,files ignored and complex files  in a csv format
+/// The structure is the following :
+/// "FILE","SIFIS PLAIN","SIFIS QUANTIZED","CRAP","SKUNK","IGNORED","IS COMPLEX","FILE PATH",
+pub fn print_metrics_to_csv<A: AsRef<Path> + Copy>(
+    metrics: Vec<FileMetrics>,
+    files_ignored: Vec<String>,
+    complex_files: Vec<FileMetrics>,
+    csv_path: A,
+    project_coverage: f64,
+) -> Result<()> {
+    debug!("Exporting to csv...");
+    export_to_csv(
+        csv_path.as_ref(),
+        metrics,
+        files_ignored,
+        complex_files,
+        project_coverage,
+    )
+}
+
+/// Prints the the given  metrics ,files ignored and complex files  in a json format
+pub fn print_metrics_to_json<A: AsRef<Path> + Copy>(
+    metrics: Vec<FileMetrics>,
+    files_ignored: Vec<String>,
+    complex_files: Vec<FileMetrics>,
+    json_output: A,
+    project_folder: A,
+    project_coverage: f64,
+) -> Result<()> {
+    debug!("Exporting to json...");
+    export_to_json(
+        project_folder.as_ref(),
+        json_output.as_ref(),
+        metrics,
+        files_ignored,
+        complex_files,
+        project_coverage,
+    )
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use core::cmp::Ordering;
+    const JSON: &str = "./data/seahorse/seahorse.json";
+    const COVDIR: &str = "./data/seahorse/covdir.json";
+    const PROJECT: &str = "./data/seahorse/";
+    const IGNORED: &str = "./data/seahorse/src/action.rs";
+
+    #[inline(always)]
```

Comment:
Let's transform this one in a utlity function because we can use it in other parts of the code